File: src/old/app/core/PropertiesBuffer.java
package app.core;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.util.Collections;
import java.util.Map;
import java.util.Properties;

public class PropertiesBuffer {
  private PropertiesBuffer() {
  }
  private static boolean checkSums(final File f) {
    return f.exists() && f.isFile() && f.canRead() && f.canWrite() && f.length() > 0;
  }
  public static Map<Object, Object> readProperties(final File f) {
    if(checkSums(f)) {
      try {
        Properties p = new Properties();
        p.load(new FileInputStream(f));
        return p;
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    return Collections.emptyMap();
  }
}

File: src/app/core/PropertiesBuffer.kt
package app.core

import app.core.PropertiesBuffer
import java.io.File
import java.util.Properties
import java.io.FileInputStream
import java.io.IOException

object PropertiesBuffer {
    private fun checkSums(f: File): Boolean {
        return f.exists() && f.isFile && f.canRead() && f.canWrite() && f.length() > 0
    }

    fun readProperties(f: File): Map<Any, Any> {
        if (checkSums(f)) {
            try {
                val p = Properties()
                p.load(FileInputStream(f))
                return p
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
        return emptyMap()
    }
}